src/sqlacodegen/generators.py

```diff
@@ -1243,9 +1243,23 @@ def render_python_type(column_type: TypeEngine[Any]) -> str:
             if isinstance(column_type, DOMAIN):
                 python_type = column_type.data_type.python_type
             else:
-                python_type = column_type.python_type
+                try:
+                    python_type = column_type.python_type
+                except NotImplementedError:
+                    self.add_literal_import("typing", "Any")
+                    python_type = Any
+
+            def get_python_type_name() -> str:
+                if hasattr(python_type, "__name__"):
+                    return python_type.__name__
+                elif isinstance(python_type, _SpecialForm):
+                    return python_type._name
+                else:
+                    raise AttributeError(
+                        f"Cannot get name from type {python_type}: no __name__ attribute and not a _SpecialForm"
+                    )
 
-            python_type_name = python_type.__name__
+            python_type_name = get_python_type_name()
```

Comment:
Why are we wrapping this logic in a nested function that only gets called once?

Reply:
Just for the sake of clarity. I can move it out of the function if you feel it is better to do so

Reply:
@agronholm I added your final suggestion. Giving it a second look, I was able to remove some additional duplicate handling